### DISCUSSION TITLE
New Part: Hitbox

### DISCUSSION THREAD

**Discussion by Logzombie** [CapsAdmin/pac3#1123] [Ideas]
Quick Disclaimer: This is not to do with the player's entity, we already have hull size and generated MDL hurtboxes.

This is essentially a redux of a previous post (#964) on a part to kill two birds with one stone, and to add one of the most requested features in PAC thus far, and a very handy replacement for projectiles in a lot of scenarios.
Essentially the whole functionality of the part is to draw an invisible cube, or sphere (tickbox setting similar to projectile parts) in an area (preferably with visual indicator while in the editor,) and deal a set amount of damage to anything inside of said area. Said shape can be positioned manually to either follow a bone, or the part's parent object (E.G a model part.) Additional options will really make this part tick as long as it has heavy customisation options, and I have suggestions for these in particular, here's what I think would work best for essentials and optional settings:

Bare minimum options:
Damage values
Damage types
Size
Scale (XYZ scaling with a nearly indefinite size cap. Allowing for extremely lengthy or large hitboxes. This also covers the HUGE request for Hitscan, since you can make a very long, extremely thin hitbox to replicate hitscan's functionality. V. Important if this idea is to be added.)
Position and Angles
Follow (Will follow the player's entity and angles while active, if unticked, will simply remain static until the part is reactivated.)

Optional stuff:
Propulsion (The ability to push props or players toward an axis, WITHOUT causing damage if possible. Since the box/sphere can be rotated, you can allow for whatever angle you want to be able to push objects, preferably with a value for Force, aka how far an object will be sent flying.)
Persistent Damage (If the part remains active, anything within the area will receive continuous damage based on a value specified for how many seconds pass. This is to allow for area of effect that may cause damage over time if it remains active without the need for timer parts resetting the part..)

A lot of this has been inspired from how other games manage their area of effects, and the infinite scaling options will allow for users to not only use this part for AoE, but also for hitscan weaponry. Considering projectile parts use _"Damage Radius"_ for an AoE on impact only, I believe this is entirely possible to implement if an area or shape can be drawn to be used as a hitbox. This can be used to entirely replace projectiles if used for melee attacks or some bullet based weaponry. Hitscan has been requested for ages now, and this is an ideal way of implementing it, as well as bringing something also very handy to the table at the same time. _This, will be stupidly useful if implemented._

**Comment by Darkcraft8** [comment]
it would be interesting if you could make it physical (with some limitation off course), it could be used to make model hitbox (could trigger a event if touched or hit for exemple: collision for model or breakable "props") or extend player hitbox (exemple: shield that block bullet and projectile without being a sphere or weak point that deal double damage).
